### PR TITLE
Botany rework V2

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
[Wiki guide here!](https://www.paradisestation.org/wiki/index.php?title=User:FunnyMan3595/Reworked_guide_to_hydroponics)

New mutation mechanics:
* Plants mutations no longer affect the growing plant or its produce, only the resulting seeds.
* Mutations are scaled by the mutation level of the plant tray, from 0 to 35.
* Mutrient gives +10 mutation level, L4Z gives +15.
* UM, radium, and uranium can be added to the tray, up to 15u total. Each 1u gives +1 mutation level. 1u is drained per harvest. You can reset this with 1u of potassium iodide or pentetic acid.
* Shooting a tray with a floral somatory in mutate mode will give +5 mutation level for the next harvest. Does not stack.
* Above a mutation level of 10, the final seed may become a different species, with higher levels increasing the chance.
* Above a mutation level of 25, a random trait may be added to the final seed, with higher levels increasing the chance.
* Above a mutation level of 30, the final seed will always become a different species if possible.
* Hydroponics trays can now be doped with six different chemicals to bias their mutation to a particular stat. Saltpetre for potency, ammonia for yield, diethylamine for production, cryoxadone for endurance, omnizine for lifespan, and Saline-Glucose for weed growth and vulnerability. Charcoal will clear the doping chemical.
* Doping increases the effective mutation level to double for the selected stat(s), while reducing it to half for other stats.

New seed mechanics:
* Seeds extracted from a plant with any mutation level will become unsorted seeds.
* An un-upgraded seed extractor can only extract one pack of unsorted seeds per plant.
* Unsorted seeds cannot be planted or stored in the seed extractor, but they can be recycled in the biogenerator.
* To get usable seeds, you must sort them:
  * Use unsorted seeds in hand. Takes 2s, minimal mutations.
  * Place unsorted seeds in a seed sorting tray, use it in hand. Takes 1s per pack, mutations increase after the 4th pack sorted at once.
  * Place unsorted seeds in any kind of container and use a plant analyzer on it. Takes 0.5s per pack, mutations increase after the 4th and 8th pack sorted at once.
* Seeds with different statistics no longer stack in a plant bag.
* If you are wearing a Hydroponics HUD, you will get a fast tooltip for seed packs, showing their statistics.

Miscellaneous changes:
* The plant analyzer has been updated to show the new information.
* Self-sufficient trays always use Earthsblood fertilizer, equivalent to EZ-Nutrient, and do not benefit from the yield modifier of bees.
* Plant yield stats go up to 10 again.
* Shooting a plant with a floral somatoray in yield mode will give it a one-time boost of 1-3 produce, ignoring all yield modifiers.
* The extended NutriMax inventory has changed to 10 bottles of ash (for healing plants), 2 bottles of saltpetre (for doping), 2 bottles of charcoal (for un-doping), and 2 bottles of unstable mutagen (for mutating plants).
* Wheat's starting potency has been boosted from 15 to 30, to make it easier to meet the chef's first need.
* Kudzu is now obtained by letting weeds overtake soybeans.

(Review note: Includes #25031 until that's merged)

## Why It's Good For The Game
All but the simplest of botany is heavily dependent on the chem dispenser, partially for UM, but even more so for the copious amounts of Saltpetre and Diethylamine you use to max out your plant stats.  It turns out, however, that dumping bucket after bucket of reagents onto a tray isn't that interesting a mechanic.

These changes remove the need for Saltpetre and Diethylamine almost entirely, and significantly reduce the amount of UM required. The new flow looks like this:
* For crops you just want to grow and use, continue to use EZ or RH.
* For crops you want to change species, like ambrosia and glowshrooms, L4Z and 15u of UM are enough to guarantee that your seeds will change species after sorting. And since the mother plant never mutates, as long as it's perennial, you can just keep harvesting if you don't get the species you wanted.
* For increasing plant stats, you can tune the amount and kind of mutation with the mutation level and doping, but even simple Mutrient is sufficient to get production 1, yield 10, and potency 100 in about 20 mins, like so:
  1. Plant a grass seed, give it Mutrient.
  2. Harvest the grass and extract its seeds.
  3. Sort the seeds, and pick out the ones with the lowest production. Plant them.
  4. Dig up any plants you no longer need.
  5. Repeat 2-4 until production 1.
  6. Repeat 2-5 with yield until 10, and potency until 100, refusing to let prior stats decay.

Grass is likely not the optimal choice anymore, as despite having a perfect maturation speed of 2, its starting production speed of 5 and yield of 5 both hamper it.

Regarding yield 10, there were a couple problems with the reduced cap:
* Some plants had >5 yield by default, and could keep it until they mutated or were gene-edited.
* It punished low-end and mid-grade botanists for the behavior of powergamers.

Since this PR already forces self-sufficient trays to use a yieldmod of 1, meaning you can't improve them with RH or bees, the only way to get more than 10 plants per harvest now is if you're manually caring for your plants, which should help deter powergamers.

## Testing
Extensive, plus testmerge.

## Changelog
:cl:
add: Completely new plant mutation mechanics. Check the PR or the wiki.
del: Plants in trays and their produce are no longer affected by mutations.
add: Unsorted seeds, produced by mutating plants. They can be converted to normal seeds in hand, with a seed sorting tray, or by scanning a container holding them. Scanning more at once is usually beneficial.
add: While wearing a Hydroponics HUD, you can hover over seeds to see their statistics.
tweak: Seeds no longer stack unless their statistics match.
tweak: Self-sufficient trays always use Earthsblood fertilizer, equivalent to EZ-Nutrient, and do not benefit from the yield modifier of bees.
tweak: Plant yield stats go up to 10 again.
tweak: The basic yield of wheat has increased from 15 to 30
tweak: The plant analyzer was updated to match the new mechanics.
tweak: Shooting a plant with a floral somatoray in yield mode will give it a one-time boost of 1-3 produce, ignoring all yield modifiers.
tweak: The extended NutriMax inventory has changed to 10 bottles of ash (for healing plants), 2 bottles of saltpetre (for doping), 2 bottles of charcoal (for un-doping), and 2 bottles of unstable mutagen (for mutating plants).
tweak: Wheat's starting potency has been boosted from 15 to 30, to make it easier to meet the chef's first need.
tweak: Kudzu is now obtained by letting weeds overtake soybeans.
/:cl:

### CODE REVIEW

**Comment by Aquilar** [comment]
why not make self sufficient trays a toggle and unlocked by part upgrade instead? 
Then we can remove the tedium of getting Gaia and prevent rookie/malicious botanists from screwing yields crops over.

**Comment by CinnamonSnowball** [comment]
> On the whole, this should make botany slower to set up, but more engaging. And once it is set up, getting stats onto a new plant is much less tedious than it used to be.

These claims don't seem to be backed by anything - in fact, getting good stats on plants seems *more* tedious now that you can't save specific high stats and preserve them, and instead have to get lucky on all stats at once, or get bent. Botany can also no longer produce their own stat-mutation inducing chemical, Unstable Mutagen, and have to bother science for a Somatoray or spend ages waiting for their plants to grow and mutate manually. This is not better or less tedious at all.

Removing the use of Saltpeter/Diethylamine seems.. very unnecessary, to say the least? They're by no means required for botany, they literally exist just to speed things up a bit if botany wants to ask for a dispenser. If you feel that botany uses the chem dispenser too much, that's entirely because of our current botany culture, not mechanical requirements, and the way to "fix" it would be to add more encouragement to the natural method (like the recent addition of glowshrooms to base seed vendor items, rather than requiring a multitool to hack it for them), rather than outright removing the "problematic" methods.

There *are* a couple good changes in this PR, but as a whole it seems very misguided and it sounds like the changes made weren't really thought through.

File: code/modules/hydroponics/gene_modder.dm [hunk 1]
```diff
@@ -402,11 +412,39 @@
...
+		name = ""
+		if(seeds_scanned < seeds_needed)
+			name +=  "[round(seeds_scanned/seeds_needed*100,1)]% of "
+		name += "Core genes "
+		ui_name = "Core "
+		for(var/i in 1 to length(core_genes))
+			if(i > 1)
+				name += "/"
+				ui_name += "/"
+			name += "[core_genes[i].value]"
```
Comment (GDNgit) [inline code/modules/hydroponics/gene_modder.dm:430]: Nonstatic type, gotta cast it first

Comment (FunnyMan3595) [reply code/modules/hydroponics/gene_modder.dm]: Done, thanks!

**Comment by FunnyMan3595** [comment]
For those who aren't convinced of the changes, **please try them**. Theorycrafting is nice, but I did my own to arrive at these changes, and I've since tried them out and find that, at least for me, they play well. I'm more than happy to listen to well-founded criticism, but practice is *far* more useful than theory.

There's a [starter wiki page](https://paradisestation.org/wiki/index.php?title=User:FunnyMan3595/Reworked_guide_to_hydroponics) up now.

Experienced botanists can mostly [skip to here](https://paradisestation.org/wiki/index.php?title=User:FunnyMan3595/Reworked_guide_to_hydroponics#Statistical_Improvement), though there's some useful tidbits earlier:

I've also got a personal test server that you can poke me for details/permissions of in Discord, and it will likely be back on the usual test server (Denghis') in the next day or two, since the other testmerge has accomplished its goal. And, of course, you can set up your own local copy.

**Comment by Komrad822** [comment]
While I agree gaseous does need to go or be nerfed, the removal of it will not save this PR from other issues I see in it.

> Bees and blumpkins make the chem dispenser irrelevant. Only used because science is impatient and its a player mentality problem, not a botany problem.

> Removing RNG with even slower RNG seems redundant. And still, you'd just be pouring chems into a plant to change stats JUST as before.

> Removing the modularity of plants by forcing all stats onto a single disk really removes more than it adds. Being able to control a plant to be just how you want it to be is nice.

**Comment by FunnyMan3595** [comment]
Based on some playtester feedback (thanks @DGamerL), I've re-added the ability to extract single genes. This now coexists with the bulk extraction method; both are otherwise unchanged.

**Comment by Screampuff** [comment]
Really, really not a fan of basically everything being even more RNG based.  Removing UM along with guaranteed stat increasing chems just makes it a miserable grind.  I don't know if I'd really like playing botany if I quite literally have no way to make my plants better other than "use L4Z on it, harvest, and pray."

Yield 5 in my experience has been more than enough to get everything done for the chef and for R&D.

As Komrad said, the chem dispenser thing is entirely a player culture problem, not a mechanics problem.  You can make blumpkins and use bees to get the same results but slower (and more engaging and fun, frankly).  This is taking a hand grenade to an anthill.

**Comment by FunnyMan3595** [comment]
So, I did [a simulation](https://colab.research.google.com/drive/1ZPgBq8IYvOqfBo9-6E9Vdw4zymYGWYPd?usp=sharing) of a crude version of [this strategy](https://paradisestation.org/wiki/index.php?title=User:FunnyMan3595/Reworked_guide_to_hydroponics#Expert_Botany).

Over 1,000 runs, assuming it takes you 30s to pick a plant, harvest, seedify, and replant all your trays:
Best: 11:40
Worst: 56:30
Average: 26:25
And this is certainly not the optimal strategy.

In other words, the randomness here is _not nearly as bad_ as people seem to think it is. And again, I've tested this in actual play, and the randomness feels _good_, not grindy.

**Comment by DGamerL** [comment]
With the disk changes, this gets a 👍 from me, glory to dispenserless botany.

**Comment by FunnyMan3595** [comment]
!merge_upstream

**Comment by Migratingcocofruit** [comment]
While the reliance of botany on chem dispensers is addressed here, the tedium of pouring chems into containers is replaced by watching grass grow. The issue here is the tedium is replaced with a lot of doing nothing for relatively short intervals. This means doing anything else during this time will interrupt you, which forces you to choose between staying AFK and improving your plants more slowly. This is like starting the game by growing 10 different Gaia plants with high weed growth.

Still the general idea of improving the plants by iterating and selecting the better ones is good, and being able to pull the full stats of a plant without restriction plays well with it. It's just that there isn't much the player can do to influence the outcome other than grow more plants to choose from, which isn't very interactive, and doesn't give them much control of the outcome.  
I think a better solution, in a general sense would be making the botany chems more readily available via botany, and for them to be applied over time, and instead of them being used on the plant. One implementation that comes to mind is adding a reservoir in which you put your mixture that then affects the way your plants mutate their stats, the rate of weed growth, and the final yield and perhaps even growth cycle rate of the system. you pump that mixture through the trays using the previously mostly unused irrigation pipes and have to adjust it over time as nutriment gets consumed. Self sustaining trays won't affect the mixture, so they have a similar function to the current state.  
This rewards a knowledgeable botanist by allowing them to more quickly reach good stats on the plants by better optimizing the mixture, and growing the correct plants at the correct time to get the chems they need, while also providing a way to affect the growth of the plants other than waiting.  

I do think keeping mutagen behavior the same for species/trait mutation is the way to go. It keeps it consistent with the rest of the game while also allowing for more rapid acquiring of traits and chems from different species to actually do research and make cool plants by combining the traits and chems of different plants.  

Gaseous Decomposition isn't the issue so much as the smoke reaction itself is. It applies all of the reagent simultaneously to every floor tile and creature in range, effectively multiplying reagents(you can actually use this to make an infinite amount of mostly anything, though thankfully it's too impractical to pull off to be useful), and it applies is to the outside and inside with some reagents being able to penetrate hardsuits just because. Then there is Facid melting masks and hardsuit helmets and Super Hairgrownium which drops your mask to the floor. It's all quite silly. But the solution isn't removing a plant trait, it is fixing the underlying reaction that it activates. 
The reagent quantity should be spread across all tiles and each affected part of each applicable occupant, and hardsuits should protect you from cyanide, and at least have some option to protect you from Facid as well.    
As for the non power gaming uses: Carpet bombing the station, Fliptonium parties, making people scream like that shopping cart filled with rubber geese using silver sulph(also healing them a bit), fun benign role playing things to do in a role playing game. That's excluding anything that would be moderately useful, or powerful since I'm not sure where the line between power gaming and not power gamins lies. 

Overall, while there are some nice qualities here I believe the overall effect on the game would be negative. It increases the time requirement for what should be the most basic part of botany, and also takes away control from the player. The removal of Gaseous I disagree with because of the reasons I stated above, and to be honest it feels shoehorned in.

**Comment by FunnyMan3595** [comment]
!merge_upstream

**Comment by FunnyMan3595** [comment]
GD re-added per internal feedback. There's changes to it planned, so we'll wait and see how that goes and decide later on if it should be kept.

**Comment by FunnyMan3595** [comment]
!merge_upstream

**Comment by Ionward** [comment]
This just reintroduces the genetics problem of doing everything right and just getting unlucky. It feels really really bad to have fully upgraded machines and have the option to extract a good stat to try and mix and match and it tells you that you get a stat that's worse than anything naturally appearing. 

You get drip fed improvements through sheer RNG and there's no choice in what you keep out of it, there's no advantage to knowing what you're doing aside maybe getting densified chems, which feels required with how little the potency will reliably get to even have any sort of effective plant. There's no real advantage except time to getting upgrades on your manipulator. This PR aims to make it more worthwhile to watch your trays but makes it the ONLY way to get stat improvements when you've got self sufficient trays, which also end up requiring more space to accommodate all the RNG-breeding stock plants, and most chefs end up asking for like half of your trays for all their requests.
Botany's got specific problems and power-creep but this just makes it actively unfun to play.

**Comment by Aquilar** [comment]
Two suggestions:
-Make self sufficient trays A toggle-able feature, be a component upgrade and not require the use of gaia. PLEASE!
-Increased the yieldmod for self sufficient trays to 3 and be affected by robust harvest and bees; Just so botanists aren't being forced to babysit the Chefs plants in addition to whatever project they're doing. And to throw botanist Antags a bone.

**Comment by Migratingcocofruit** [comment]
I've already given my input before the test merge and after having played with this a bit it is affirmed. Playing as a botanist now feels like starting the game in Perma. Grinding for stats was never the fun part of Botany, and now it is both longer and more tedious. You need to spam even more buckets but now it's all just water and nutrient, the effect is random and you need to wait for it.  
It's more tedium and less engagement, both with the plants and everything else in the round.  
The one positive is the core stats disk, but it doesn't at all make up for the rest.

**Comment by Screampuff** [comment]
I played it myself and I'm just not having fun anymore.  Having to rely on the whims of RNG to provide critical components to RND or really, well, do much of anything besides provide the chef some food is just an absolute slog to play.

**Comment by Ionward** [comment]
After trying a bit, the second set of changes doesn't really fix anything in regards to speed or randomness now, or the other problems the first draft had. Its needlessly complicated, still just as random trying to pray that you get the stats you want, and its even slower now having to wait for a plant to harvest, pick up every thing and turn it into seeds and sort all those and then again, praying you got lucky with the random species mutation. Before the only RNG was what plant you got, now its a gamble if it even mutates. 

The plant analyzer is worse than a health one with how much info it spits at you all at once. Mutating a new species is overly complicated. You still need loads of space to try and save time getting good stats. Ultimately I don't know what the goal is except making botany so annoying to play that no one bothers with it anymore, it still doesn't really reward you for keeping your plants attended, it just makes it impossible to do anything good with them if you have them self sufficient. People still get chem dispensers because they need all these chems, and you have to gamble on there being a medchemist there that knows what they're doing and wants to waste their time getting you stuff that doesn't do much of anything in the first place.

**Comment by FunnyMan3595** [comment]
> I tried it for a shift and I like it a lot better than before. I gave my feedback on the Discord thread, but for a paper trail, feedback would be nice for UM and doping so the player knows what's happening, along with a better description on how to sort seeds out besides by hand.

Feedback has been added, will be present once the TM updates.

**Comment by Ionward** [comment]
> With seed sorting, you have multiple chances to get a mutation to happen as long as your seed extractor gets upgraded to spit out multiple unsorted seeds. This stacked with UM makes it a lot more predictable to do.

It doesn't really though, and when the game gives you so much info but the guide and game explain it poorly, it feels like an even worse gamble than before. Instead of just getting plant mutations out of the way and focusing on growing ones to harvest for traits you have to spend a bunch of time reharvesting and praying that the mutation builds. All the random types of unsorted seeds too just needlessly clutters the seed harvester when 99% of them are garbage. It also clutters the vendors and room itself with all the chemicals you now *require* to do anything. Changes like this make the DNA vault even more annoying to do than before.
And in the end I don't want to have to babysit my plants for most of the round to get them up to where they would've been in 15 minutes before. 

Again, ultimately, what is this rework even trying to fix? Its still worse to play than the normal version, with all the RNG its worse than old genetics was, and it just makes it a boring stat grind if you do get lucky. Hell, even the ghost roles don't feel like their upgrades actually do anything now.

**Comment by FunnyMan3595** [comment]
> It doesn't really though, and when the game gives you so much info but the guide and game explain it poorly, it feels like an even worse gamble than before. Instead of just getting plant mutations out of the way and focusing on growing ones to harvest for traits you have to spend a bunch of time reharvesting and praying that the mutation builds.

The guide is pretty explicit that if you use 15u of UM and Left 4 Zed, you're guaranteed to get species mutations. The game will also indicate it in the mutation level as "(will change species)" if you hit 30.

> All the random types of unsorted seeds too just needlessly clutters the seed harvester when 99% of them are garbage.

Seed extractor definitely needs a UI upgrade to handle seed types better. That was already a bit of a problem before, but this made it worse.

> It also clutters the vendors and room itself with all the chemicals you now _require_ to do anything.

You only really need EZ or RH, L4Z, Mutrient, and UM. The rest of the chems are completely optional.

> Changes like this make the DNA vault even more annoying to do than before.

Not really? I've never bothered upgrading plants for DNA Vault, there are plenty of seeds in the vendor to do it without mutating species at all, and they'll work perfectly fine in self-sufficient trays.

> And in the end I don't want to have to babysit my plants for most of the round to get them up to where they would've been in 15 minutes before.

If it's taking you more than 40 minutes to get stats fully upgraded, you're either distracted or doing something wrong. My test run took 23 minutes, and it could easily have been faster.

> Again, ultimately, what is this rework even trying to fix?

It's intended to be more fun to play, and for most people, it seems like V2 is managing that.

> Its still worse to play than the normal version, with all the RNG its worse than old genetics was, and it just makes it a boring stat grind if you do get lucky.

The RNG is very manageable now, since you should get plenty of unsorted seeds to work with -- one per harvest for species mutations (with L4Z), and one per yield per harvest for stats mutations (with Mutrient).

> Hell, even the ghost roles don't feel like their upgrades actually do anything now.

* Seed extractor gets you more seeds, especially for mutated plants.
* DNA manipulator works as before, plus reduces the number of seeds for an all-core-stats disk. Getting it down to 1 is really nice, because you don't need to regrow the final mutated seed.

**Comment by Migratingcocofruit** [comment]
Compared to V1, it is overall better, though not entirely so:  

- The mutation affecting the seeds is mostly better, as it gives you more dice rolls to get what you want. 
- The seed sorting mechanic serves no purpose other than to make you wait while standing in place. It's a worse version of making diethylamine with an igniter. It has no redeeming qualities and I think it needs to go. 
- Being able to dope the trays is good; a plain improvement over V1. However, I don't like the base mechanic to begin with.

Compared to old botany, I think it's overall worse but also offers improvements:
- Replacing the stat improvement method of chems with waiting for RNG is just worse, with or without doping. It's not even necessarily slower, speed wasn't that big of an issue even in V1, it's simply very very unfun. You literally need to watch grass grow, then repeatedly perform the same a short series of tasks each time it grows,  any interaction with the any other part of the round will slow you down at best, and you have little control over the process other than planting more plants to roll more dice. Whether or not you get through it quickly doesn't really matter, it just feels bad to play.
Additionally it makes a huge amount of unique seeds that clutter the seed extractor(which could use sorting/searching functions to begin with) to an extreme degree. 
The old mechanics for stats are alright, the only real issue was the chem dispenser.
- As stated before **sorting seeds hurts my soul.** 
- The loss of nutriment effects on perennial trays does bring some concerns as it means whatever plants you were working on would likely die if you happened to get kidnapped, died or were otherwise indisposed. It punishes you for leaving hydroponics voluntarily or otherwise. And besides, past the initial setup phase the utter majority of your time should be spent experimenting and making new plants rather than taking care of existing ones. I think having them lose production modifiers is enough. 
- Unstable mutagen was RNG to begin with, and the new way it works along with nutriment is more nuanced. I like it. You now have to sit through more maturation timers, which you cannot influence, to get to plants that require more than one mutation, like Gaia or Bluespace Tomatoes, but you do get the advantage of getting multiple types of mutations from the same plant which is very welcome and actually reduces the amount of effect that RNG has previously had. Overall a good change. One issue is, again, the cluttering of the seed-extractor, though that can be solved in its UI.


Aside from that, looking at the code there is this one odd thing:
```
if(prob(level - 50))
		add_random_traits(1, 1)
```
Considering the maximum level of 35, wouldn't that mean the new trait chance is has maximum of negative 15 percent? shouldn't this be (level-25) as per the description? After playing for a while I couldn't get any new random trait even after many harvests at 35 mutation level.

**Comment by FunnyMan3595** [comment]
> * The seed sorting mechanic serves no purpose other than to make you wait while standing in place. It's a worse version of making diethylamine with an igniter. It has no redeeming qualities and I think it needs to go.
> [...]
> ```
> if(prob(level - 50))
> 		add_random_traits(1, 1)
> ```
> 
> Considering the maximum level of 35, wouldn't that mean the new trait chance is has maximum of negative 15 percent? shouldn't this be (level-25) as per the description? After playing for a while I couldn't get any new random trait even after many harvests at 35 mutation level.

You're missing a key mechanic in both cases: sorting more seeds at once strengthens your mutations. 1.5x for the 5th-8th seed, and 2x for seed 9+ (with plant analyzer only).

> * Replacing the stat improvement method of chems with waiting for RNG is just worse, with or without doping. It's not even necessarily slower, speed wasn't that big of an issue even in V1, it's simply very very unfun. You literally need to watch grass grow, then repeatedly perform the same a short series of tasks each time it grows,  any interaction with the any other part of the round will slow you down at best, and you have little control over the process other than planting more plants to roll more dice. Whether or not you get through it quickly doesn't really matter, it just feels bad to play.

So far, most folks seem to disagree with you on this point. So it might just not be for you, which is regrettable, but fine: not every mechanic will please everyone.

>   Additionally it makes a huge amount of unique seeds that clutter the seed extractor(which could use sorting/searching functions to begin with) to an extreme degree.

Yeah, improving that is on my to-do list. In the meantime, I recommend sorting with the Hydroponic HUD's tooltips and dumping the seeds you don't want into the biogenerator.

> * As stated before **sorting seeds hurts my soul.**

It already goes at 0.5s/seed when using a plant analyzer, which is pretty fast. And if you're using the Hydroponic HUD, you can start looking for the seeds you want as the rest are sorting.

> * The loss of nutriment effects on perennial trays does bring some concerns as it means whatever plants you were working on would likely die if you happened to get kidnapped, died or were otherwise indisposed. It punishes you for leaving hydroponics voluntarily or otherwise. And besides, past the initial setup phase the utter majority of your time should be spent experimenting and making new plants rather than taking care of existing ones. I think having them lose production modifiers is enough.

So use self-sufficient trays? They still work just fine. The only time you need normal trays is when you're breeding for new plants, Stats can be done entirely with mutagen. It effectively just brings you down to 10 yield per tray, which is the same as the current intended 5 max with x2 from bees.

**Comment by Migratingcocofruit** [comment]
>So far, most folks seem to disagree with you on this point. So it might just not be for you, which is regrettable, but fine: not every mechanic will please everyone. 

Who are most folks, what did they say, where did they say that? I fail to see the commentary here, and can't seem to find it on the discord. 

>It already goes at 0.5s/seed when using a plant analyzer, which is pretty fast. And if you're using the Hydroponic HUD, you can start looking for the seeds you want as the rest are sorting.

My issue here is that the timer exists to begin with. The game shouldn't waste your time with tedious tasks, and certainly not lock you in place with a timer. Seed sorting should just happen upon extraction.

>So use self-sufficient trays? They still work just fine. The only time you need normal trays is when you're breeding for new plants, Stats can be done entirely with mutagen. Otherwise, it effectively just brings you down to 10 yield per tray, which is the same as the current intended 5 max with x2 from bees.

Mutating now requires that you wait the maturation time, which means you'd need to keep up with the plants for a few minutes until they are harvestable. This is fine for the beginning of the shift but I think after the initial setup phase it's mostly an annoyance.

**Comment by Migratingcocofruit** [comment]
>You're missing a key mechanic in both cases: sorting more seeds at once strengthens your mutations. 1.5x for the 5th-8th seed, and 2x for seed 9+ (with plant analyzer only).

image

Clicking the seeds with the analyzer while they are in a container gave me this. 
What am I doing wrong here?

**Comment by Screampuff** [comment]
Click the container itself

-------- Original Message --------

**Comment by Aleyck** [comment]
Given the poor history of comments on this PR, I feel like I now need to give updated feedback. 

I very much enjoy how botany works now. There is a feeling of nearly-assured progress, with some trial and error. The ability to see the seed stats at a glance gives the hydroponic HUD just a bit more kick (Now we just need Hydroponic Sunglasses, haha)

Doping is a bonus, mostly, not needed at all. Probably most useful with Ammonia. The do-after for sorting seeds can be taken or left, in my opinion. I have no issue with it, but I can see why someone might nitpick about that. UI for the seed machine is definitely going to need help after this merges.

**Comment by FunnyMan3595** [comment]
> UI for the seed machine is definitely going to need help after this merges.

UI has been helped, see screenshot in #25031 (which is included in this TM). Just waiting on an HC to push the new version.

**Comment by FunnyMan3595** [comment]
Tagging DNM until #25031 goes in.

**Review by S34NW — APPROVED** [review]
It's been fine enough on TM, lets go

**Pull request merged**